libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-label.pipe.ts

```diff
@@ -0,0 +1,47 @@
+import { Pipe, PipeTransform, inject } from '@angular/core';
+
+import { SkyDatepickerCalendarInnerComponent } from './datepicker-calendar-inner.component';
+
+/**
+ * Formats date values according to formatting rules.
+ * @example
+ * ```markup
+ * {{ myDate | skyDatepickerCalendarLabel }}
+ * {{ myDate | skyDatepickerCalendarLabel:'YYYY' }}
+ * ```
+ * @internal
+ */
+@Pipe({
+  name: 'skyDatepickerCalendarLabel',
+  standalone: true,
+})
+export class SkyDatepickerCalendarLabelPipe implements PipeTransform {
+  #datepicker = inject(SkyDatepickerCalendarInnerComponent);
+  #defaultFormat = this.#datepicker.formatDayLabel;
+  #format: string | undefined;
+  #formattedValue: string | undefined;
+  #value: any;
```

Comment:
Since we only assign this value if it's `instanceof Date` could we make this `Date | undefined`?

Reply:
Trying to make this `| undefined` is causing problems when we try to use it in the formatting function. It occurs to me that having these stateful variables makes this pipe implementation not properly pure, and if a consumer attempted to pass an empty date value, they would just get back the previous formatted string. Instead, I think it's better if we just streamline this into an inline function that always returns the value expected.